### DISCUSSION TITLE
How can I parse environment variable in my `body`?

### DISCUSSION THREAD

**Discussion by elect86** [ncipollo/release-action#62] [Q&A] [answered]
As titled, I'd like to know how I can parse github envs in `body`, that is
```
      - name: 🛎️ Create release
        if: strategy.job-index == 0
        uses: ncipollo/release-action@v1
        with:
          artifacts: ~/jdk-linux-x64${{ matrix.artifact }}/jdk-${{ env.JDK_VERSION }}-internal+0_linux-x64_bin${{ matrix.artifact }}.tar.gz
          token: ${{ secrets.GITHUB_TOKEN }}
          tag: $GITHUB_ACTOR-$GITHUB_SHA
          body: ${$GITHUB_ACTOR-$GITHUB_SHA}
          draft: true
```

**Comment by ncipollo** [comment, marked as answer]
Hi!

As far as I know, there shouldn't be anything special about the inputs to this action. The typical workflow rules for inputs and environment variables should apply.

In the above example, I think the issue would be that you are trying to reference environment variables directly in the workflow file (ex- ` tag: $GITHUB_ACTOR-$GITHUB_SHA`). Instead, I think you want to be accessing these variables via a context. Environment variables should only be used within the runner itself (so in shell scripts, actions, etc). Checkout the following link: https://docs.github.com/en/actions/reference/environment-variables#determining-when-to-use-default-environment-variables-or-contexts

So to summarize -  Instead of `$GITHUB_ACTOR` I think you could do something like `${{ env.GITHUB_ACTOR }}`. In this very specific case I think you could access these values via the github context: https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context

Not sure how easy it is to compose context values together like you are doing (`"${{ github.action }}-${{ github.sha }}"`),  if you can't do that via the context syntax you can create an environment variable in a previous step via these instructions: https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable.

**Reply by elect86** [reply]
Thanks Nick, that made it!

**Reply by elect86** [reply]
ps: what about using instead a command output? Such as `git describe --tags`?

**Reply by ncipollo** [reply]
If you want to pipe those into the body, I would actually use `bodyFile` instead. Example: https://github.com/ncipollo/actions-playground/blob/master/.github/workflows/tag.yml#L26

That way you could have a shell script step which does `git describe --tags >> body.md`, then set `bodyFile` to `body.md`.
